src/GPUActivityAgent.cpp

```diff
@@ -0,0 +1,453 @@
+        geopm_time(&m_last_wait);
+    }
+
+    // Adds the wait time to the top of the report
+    std::vector<std::pair<std::string, std::string> > GPUActivityAgent::report_header(void) const
+    {
+        return {{"Wait time (sec)", std::to_string(M_WAIT_SEC)}};
+    }
+
+    // Adds number of frquency requests to the per-node section of the report
+    std::vector<std::pair<std::string, std::string> > GPUActivityAgent::report_host(void) const
+    {
+        std::vector<std::pair<std::string, std::string> > result;
+
+        result.push_back({"GPU Frequency Requests", std::to_string(m_gpu_frequency_requests)});
+        result.push_back({"GPU Clipped Frequency Requests", std::to_string(m_gpu_frequency_clipped)});
+        result.push_back({"Resolved Max Frequency", std::to_string(m_f_max)});
+        result.push_back({"Resolved Efficient Frequency", std::to_string(m_f_efficient)});
+        result.push_back({"Resolved Frequency Range", std::to_string(m_f_range)});
+
+        for (int domain_idx = 0; domain_idx < M_NUM_GPU; ++domain_idx) {
+            double energy_stop = m_gpu_active_energy_stop.at(domain_idx);
+            double energy_start = m_gpu_active_energy_start.at(domain_idx);
+            double region_stop = m_gpu_active_region_stop.at(domain_idx);
+            double region_start =  m_gpu_active_region_start.at(domain_idx);
+            result.push_back({"GPU " + std::to_string(domain_idx) +
+                              " Active Region Energy", std::to_string(energy_stop - energy_start)});
+            result.push_back({"GPU " + std::to_string(domain_idx) +
+                              " Active Region Time", std::to_string(region_stop - region_start)});
+        }
+
+        return result;
+    }
+
+    // This Agent does not add any per-region details
+    std::map<uint64_t, std::vector<std::pair<std::string, std::string> > > GPUActivityAgent::report_region(void) const
+    {
+        return {};
+    }
+
+    // Adds trace columns signals of interest
+    std::vector<std::string> GPUActivityAgent::trace_names(void) const
+    {
+        return {};
+    }
+
+    // Updates the trace with values for signals from this Agent
+    void GPUActivityAgent::trace_values(std::vector<double> &values)
+    {
+    }
+
+    std::vector<std::function<std::string(double)> > GPUActivityAgent::trace_formats(void) const
+    {
+        return {};
+    }
+
+    // Name used for registration with the Agent factory
+    std::string GPUActivityAgent::plugin_name(void)
+    {
+        return "gpu_activity";
+    }
+
+    // Used by the factory to create objects of this type
+    std::unique_ptr<Agent> GPUActivityAgent::make_plugin(void)
+    {
+        return geopm::make_unique<GPUActivityAgent>();
+    }
+
+    // Describes expected policies to be provided by the resource manager or user
+    std::vector<std::string> GPUActivityAgent::policy_names(void)
+    {
+        return {"GPU_FREQ_MAX", "GPU_FREQ_EFFICIENT", "GPU_PHI", "SAMPLE_PERIOD"};
```

Comment:
```suggestion
        return {"GPU_FREQ_MAX", "GPU_FREQ_EFFICIENT", "GPU_PHI"};
```
Post 2.0 Agents are expected to use a more general geopmlaunch option to set sample period/control rate.  SAMPLE_PERIOD should be removed from the agent and supporting collateral